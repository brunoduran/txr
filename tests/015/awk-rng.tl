(awk (:inputs '#"X1 a b X1 c X2 1 2 3 Y1 r s Y2 t Y2"
              '#"X1X2 Y1Y2"
              '#"X1X2 Y1 a Y2 X1 b X2")
     (:set ofs "\t")
     (:begin
       (prn 'rec 'rng '-rng 'rng- '-rng- '--rng- '--rng 'rng+ '-rng+ '--rng+ 'Y1Y2))
     (t (prn rec
             (rng (rng #/X1/ #/X2/) (rng #/Y1/ #/Y2/))
             (-rng (rng #/X1/ #/X2/) (rng #/Y1/ #/Y2/))
             (rng- (rng #/X1/ #/X2/) (rng #/Y1/ #/Y2/))
             (-rng- (rng #/X1/ #/X2/) (rng #/Y1/ #/Y2/))
             (--rng- (rng #/X1/ #/X2/) (rng #/Y1/ #/Y2/))
             (--rng (rng #/X1/ #/X2/) (rng #/Y1/ #/Y2/))
             (rng+ (rng #/X1/ #/X2/) (rng #/Y1/ #/Y2/))
             (-rng+ (rng #/X1/ #/X2/) (rng #/Y1/ #/Y2/))
             (--rng+ (rng #/X1/ #/X2/) (rng #/Y1/ #/Y2/))
             (rng #/Y1/ #/Y2/))))

