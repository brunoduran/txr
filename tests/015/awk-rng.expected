rec	rng	-rng	rng-	-rng-	--rng-	--rng	rng+	-rng+	--rng+	Y1Y2
X1	t		t				t			
a	t	t	t	t			t	t		
b	t	t	t	t			t	t		
X1	t	t	t	t			t	t		
c	t	t	t	t			t	t		
X2	t	t	t	t			t	t		
1	t	t	t	t	t	t	t	t	t	
2	t	t	t	t	t	t	t	t	t	
3	t	t	t	t	t	t	t	t	t	
Y1	t	t				t	end	t	t	t
r							end	end	t	t
s							end	end	t	t
Y2							end	end	t	t
t										
Y2										
X1X2	t		t				t			
Y1Y2	t	t				t	end	t	t	t
X1X2	t		t				t			
Y1	t	t				t	end	t	t	t
a							end	end	t	t
Y2							end	end	t	t
X1X2Y1Y2	t					t	end	end	t	t
a										
X1X2Y1Y2	t					t	end		t	t
X1X2Y1Y2	t					t	end	end	t	t
X1	t		t				t			
b	t	t	t	t			t	t		
X2	t	t	t	t			t	t		
